vb.go

```diff
@@ -0,0 +1,54 @@
+package sgbucket
+
+import (
+	"fmt"
+	"sync"
+)
+
+type VbucketNo uint32
+type VbucketSeq uint64
+
+// Per-vbucket sequence counter
+type VbucketSeqCounter interface {
+	Incr(vbNo VbucketNo) (VbucketSeq, error)
+	Get(vbNo VbucketNo) (VbucketSeq, error)
+}
+
+type mapVbucketSeqCounter struct {
+	data        map[VbucketNo]VbucketSeq
+	mu          *sync.Mutex
+	numVbuckets int
+}
+
+func NewMapVbucketSeqCounter(numVbuckets int) *mapVbucketSeqCounter {
+	return &mapVbucketSeqCounter{
+		data:        map[VbucketNo]VbucketSeq{},
+		mu:          &sync.Mutex{},
+		numVbuckets: numVbuckets,
+	}
+}
+
+func (msq mapVbucketSeqCounter) Incr(vbNo VbucketNo) (VbucketSeq, error) {
+	msq.mu.Lock()
+	defer msq.mu.Unlock()
+
+	if vbNo > VbucketNo(msq.numVbuckets-1) {
```

Comment:
Yeah, good point, I've been confused by that too. 

This might be clearer:

```
type VbucketNo uint32

func (v VbucketNo) InRange(maxVbNo VbucketNo) bool {
	return v < maxVbNo
}

func main() {
	var vbNo VbucketNo
	vbNo = 4
	inRange := vbNo.InRange(2)
	fmt.Printf("inRange: %v\n", inRange)
}
```

[^^ run that code on go playground](https://play.golang.org/p/SlMLn0CdyR)